### ISSUE TITLE
[BUG] [StartMenu] Wrong sorting in submenus of "custom" items pointing to folders

### ISSUE THREAD

**Issue by VitRom** [Open-Shell/Open-Shell-Menu#356]
Subj and this order is crazy.

The folder on screenshot is a bunch of misc. tools gathered over several years. With a wide range of "modified" and "created" date-time stamps. And with a few dirs named starting with dot.

Firstly it looks like a kind of "counter overflow" when menu populated firstly as it should then sees these dot-dirs and starts enumeration again... But! For example the _BITSAdmin-XP.EXE_ (3rd column) timestamps are older then all of these dot-dirs, both creation and modification.

For a several month I'm still unable to change this order somehow nor to find any logic causing such an order.

**Expected behaviour**
A clear and concise sorting order. Similar to the rest of OS shell highly preferred.

**Screenshots**
![image](https://user-images.githubusercontent.com/453142/82751130-dff55b00-9dbd-11ea-8b7d-9fa4e082c9ef.png)

**Version:**
 - Open-Shell: 4.4.131
 - OS: Win7SP1-64

**Comment by VitRom** [comment]
Just checked for a kind of some internal counter overflow.
Looks like a false assumption because one folder "wraps" after 66 items and another at 58.
But still suspect something like it anyway.
And removing cache DB from AppData doesn't help.

![wrapping-1](https://user-images.githubusercontent.com/453142/89731163-924bce80-da4d-11ea-97b7-d157c76b4b06.png) ![wrapping-2](https://user-images.githubusercontent.com/453142/89731167-9aa40980-da4d-11ea-97ee-9dee6f77c796.png)

**Comment by VitRom** [comment]
Well, something buggy still inside of internal sorting routines. Even after upgrading to 4.4.152 all stood the same. Playing with these "runned away" follders I've renamed a pair of them, replaced 1st dot with another markers, plus and underscore. And they immediately started a new "section" with a few files after.

But uninstall-reboot-cleanup-install did job and now, after such an "upgrade", the sorting is as it should.

**Comment by GH-User-X** [comment]
Did you still have all your files/folders after you reinstalled? I'm having sort of the same issue.

**Comment by GH-User-X** [comment]
Also, how does the cleanup part work?

**Comment by VitRom** [comment]
> Did you still have all your files/folders after you reinstalled? Also, how does the cleanup part work?

If You mean these folders linked into Custom items with buggy menus then I doesn't touched them at all.

"Cleanup" is too loud. I just exported config and uninstalled program. Then logged out and back in and removed everything related found in both %AppData%\Roaming and ...\Local and registry. Just to be sure. Next installed program as always and made a new Custom item pointed onto the one of problematic dirs. Everything was OK, so I restored the previously exported config.

May be it's enough to wipe %AppData%\Local only, but IIRC removal of DataCache.dbd only did nothing for me.
